test/terra/backends/aer_simulator/test_initialize.py

```diff
@@ -146,3 +147,22 @@ def test_initialize_sampling_opt_disabled(self, method, device):
         self.assertSuccess(result)
         sampling = result.results[0].metadata.get('measure_sampling', None)
         self.assertFalse(sampling)
+
+    @supported_methods(SUPPORTED_METHODS)
+    def test_initialize_with_labels(self, method, device):
+        """Test sampling optimization"""
+        backend = self.backend(method=method, device=device)
+
+        #expected
+        circ = QuantumCircuit(4)
+        circ.initialize(Statevector.from_label('+-rl'))
+        circ.save_statevector()
+        expected = backend.run(circ).result().get_statevector(circ)
+
+        #actual
+        circ = QuantumCircuit(4)
+        circ.initialize('+-rl')
+        circ.save_statevector()
+        actual = backend.run(circ).result().get_statevector(circ)
+
+        self.assertAlmostEqual(actual, expected)
```

Comment:
It probably wouldn't hurt just to compare against the actual target statevector, rather than passing the expected value through its own simulation - we know exactly what it should look like already.